pkg/core/blockchain.go

```diff
@@ -552,6 +559,34 @@ func (bc *Blockchain) storeBlock(block *block.Block) error {
 					}
 					break
 				}
+
+				tx, _, err := cache.GetTransaction(input.PrevHash)
+				if err != nil || int(input.PrevIndex) > len(tx.Outputs) {
+					return err
+				}
+				acc, err := cache.GetAccountState(tx.Outputs[input.PrevIndex].ScriptHash)
+				if err != nil {
+					return err
+				}
+
+				var changed bool
+				for i := range acc.Unclaimed {
+					if acc.Unclaimed[i].Tx == input.PrevHash && acc.Unclaimed[i].Index == input.PrevIndex {
+						copy(acc.Unclaimed[i:], acc.Unclaimed[i+1:])
+						acc.Unclaimed = acc.Unclaimed[:len(acc.Unclaimed)-1]
+						changed = true
+						break
+					}
+				}
+
+				if !changed {
+					bc.log.Warn("no spent coin in the account",
```

Comment:
Logging TX hash would be very useful here.

Reply:
done